### PR TITLE
feat: add fixture collecting script

### PR DESCRIPTION
**Pull request by joostlek**

### CODE REVIEW

**Comment by bdraco** [comment]
Maybe we should move this and `experiment.py` into `devtools`?

**Comment by cgarwood** [comment]
Lets add the following endpoints as well:
```
/ivp/ensemble/inventory
/ivp/ensemble/power
/ivp/ensemble/dry_contacts
/ivp/ss/dry_contact_settings
```

File: devtools/fixture_collector.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    await envoy.setup()
+
+    username = os.environ.get("ENVOY_USERNAME")
+    password = os.environ.get("ENVOY_PASSWORD")
+    token = os.environ.get("ENVOY_TOKEN")
+
+    await envoy.authenticate(username=username, password=password, token=token)
+
+    try:
+        os.mkdir("enphase")
```
Comment (bdraco) [inline devtools/fixture_collector.py:23]: Maybe make the dir named `enphase-{firmware_version}`?

File: devtools/fixture_collector.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        "/ivp/ensemble/inventory",
+        "/ivp/ensemble/dry_contacts",
+        "/ivp/ss/dry_contact_settings",
+    ]
+
+    for end_point in end_points:
+        try:
+            json_dict = await envoy.request(end_point)
+        except Exception as e:
+            continue
```
Comment (bdraco) [inline devtools/fixture_collector.py:43]: You'll have to add a `# nosec` to silence bandit

File: devtools/fixture_collector.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    await envoy.setup()
+
+    username = os.environ.get("ENVOY_USERNAME")
+    password = os.environ.get("ENVOY_PASSWORD")
+    token = os.environ.get("ENVOY_TOKEN")
+
+    await envoy.authenticate(username=username, password=password, token=token)
+
+    try:
```
Comment (bdraco) [inline devtools/fixture_collector.py:22]:
```suggestion
    target_dir = f"enphase-{envoy.firmware}"
    try:
```

File: devtools/fixture_collector.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        os.mkdir(f"enphase-{envoy.firmware}")
```
Comment (bdraco) [inline devtools/fixture_collector.py:23]:
```suggestion
        os.mkdir(target_dir)
```

File: devtools/fixture_collector.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        "/ivp/ss/dry_contact_settings",
+    ]
+
+    for end_point in end_points:
+        try:
+            json_dict = await envoy.request(end_point)
+        except Exception:
+            continue  # nosec
+        file_name = end_point[1:].replace("/", "_")
+        with open(f"enphase/{file_name}") as fixture_file:
```
Comment (bdraco) [inline devtools/fixture_collector.py:45]:
```suggestion
        with open(f"{target_dir}/{file_name}") as fixture_file:
```

but use os.path.join instead

**Review by bdraco — APPROVED** [review]

**Pull request merged**